doc/interoperability/pandas.rst

```diff
@@ -0,0 +1,299 @@
+
+
+At the time of writing, it is the users responsibility to ensure that the
+column ordering in the data used for training the estimator is the same as the
+ordering of the data used for prediction. There is an ongoing discussion
+whether or not this will change in the future and this
+`issue <https://github.com/scikit-learn/scikit-learn/issues/7242>`__ should be
+watched and used to update this paragraph in the future. A simple and straight-
+forward way of ensuring that column ordering and column labels are the same is
+using something like :meth:`df.loc[:, list of column names]` to enforce the
+correct ordering.
+
+Handling Categorical data
+=========================
+
+Section to be extended.
+
+See the following references to get started:
+
+- https://scikit-learn.org/stable/glossary.html#term-categorical-feature
+- https://scikit-learn.org/stable/modules/preprocessing.html#preprocessing-categorical-features
+- https://github.com/scikit-learn-contrib/sklearn-pandas
+
+
+Dealing with heterogenous data
+==============================
+
+Many modern datasets used with Scikit-learn contain heterogenous data. For the
+purpose of adding bespoke preprocessing steps for separate columns, Scikit-
+learn provides an experimental :class:`~sklearn.compose.ColumnTransformer` API.
+This API (which might change in the future) allows the definition of different
+transformation steps to be applied to different columns in either arrays,
+sparse matrices or pandas DataFrames.
+
+Dealing with missing values
+===========================
+
+As per the glosary, most Scikit-learn estimators do not work with missing
+values. If they do, NaN is the preferred representation of missing values. For
+more details, see https://scikit-learn.org/stable/glossary.html#term-missing-values.
+
+
+Sparse DataFrames Handling
+=============================
+
+.. note::
+  **Issue:**
+  ``Sparse DataFrames`` are not automatically converted to ``scipy.sparse``
+  matrices.
+
+This is an issue which has vastly improved from pandas version 0.21.1 onwards.
+The conversation from DataFrames has been largely optimised and are much faster
+to convert.
+
+In general, Sparse data structures (i.e. DataFrames, Series, Arrays) are memory
+optimised structures of their standard counterparts. They work on the principle
+that they contain a lot of NaN, 0, or another repeating value (this can be
+specified), and as such a lot of memory can be saved, which means one can
+potentially work with datasets that would otherwise be too large to fit into
+available memory. However one has to be careful they don't get converted into
+the dense format by mistake.
+
+In Pandas, the main sparse data structures is: :class:`~pandas.SparseArray`.
+Both :class:`~pandas.SparseDataFrame` and :class:`~pandas.SparseSeries` exists
+but don't offer significant advantages over DataFrames where the columns are
+SparseArrays and will be deprecated at some point in the future.
```

Comment:
I would be a bit more vague here. We don't know the exact plan what we are going to do in pandas. So I would keep to the current situation here, and I think the most important point that scikit-learn users need to know is: scikit-learn does not support sparse pandas structures, by default they will also be converted to numpy arrays (and thus densified). If you want to keep it sparse, you need to convert to a sparse Scipy matrix yourself manually. (is that correct? I personally never used it)